### PR TITLE
Add and test mechanism for creating graph with edge index as weight

### PR DESCRIPTION
**Pull request by ChuckHastings**
Missed this in the last uniform neighborhood sampling updates.

Added a mechanism for creating a graph using an edge index as the weight.  This is a temporary mechanism until we add direct support for edge indexes in the graph.

### CODE REVIEW

File: cpp/tests/c_api/mg_test_utils.cpp [hunk 1]
```diff
@@ -191,3 +191,102 @@ extern "C" int create_mg_test_graph(const cugraph_resource_handle_t* handle,
 
   return test_ret_value;
 }
+extern "C" int create_mg_test_graph_with_ids(const cugraph_resource_handle_t* handle,
```
Comment (seunghwak) [inline cpp/tests/c_api/mg_test_utils.cpp:194]: `with_edge_ids` might be more intuitive?

**Review by seunghwak — APPROVED** [review]
LGTM.

File: cpp/tests/c_api/mg_test_utils.cpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/tests/c_api/mg_test_utils.cpp]: Fixed in latest push

**Review by jnke2016 — APPROVED** [review]
Reviewed and tested. It looks good to me

**Comment by BradReesWork** [comment]
@gpucibot merge
